application/client.core/src/app/environment/services/service.file.opener.ts

```diff
@@ -69,47 +69,55 @@ export class FileOpenerService implements IService, IFileOpenerService {
     }
 
     public open(files: IFile[]) {
-        if (files.length === 0) {
-            return;
-        }
         if (files.length === 0) {
             return;
         }
         this._setSessionForFile().then((session: ControllerSessionTab) => {
             TabsSessionsService.setActive(session.getGuid());
-            if (files.length === 1) {
-                // Single file
-                ServiceElectronIpc.request(new IPCMessages.FileOpenRequest({
-                    file: files[0].path,
-                    session: session.getGuid(),
-                }), IPCMessages.FileOpenResponse).then((response: IPCMessages.FileReadResponse) => {
-                    if (response.error !== undefined) {
-                        this._logger.error(`Fail open file "${files[0].path}" due error: ${response.error}`);
-                        // TODO: add notification here
-                        return;
-                    }
-                }).catch((error: Error) => {
-                    this._logger.error(`Fail open file "${files[0].path}" due error: ${error.message}`);
-                    // TODO: add notification here
-                });
-            } else {
-                // Multiple files
-                // Select way: merge or concat
-                PopupsService.add({
-                    id: 'opening-file-dialog',
-                    caption: `Opening files`,
-                    component: {
-                        factory: DialogsMultipleFilesActionComponent,
-                        inputs: {
-                            files: files
+            ServiceElectronIpc.request(new IPCMessages.FileCheckRequest({
+                files: files.map((file: IFile) => file.path),
+                guid: session.getGuid(),
+            }), IPCMessages.FileCheckResponse).then((checkResponse: IPCMessages.FileCheckResponse) => {
```

Comment:
If there are no empty session `_setSessionForFile`  will create a new session, but it `IPCMessages.FileCheckRequest` will return error; session will stay empty. 

From this point of view, better call `_setSessionForFile` after a list of files was gotten. I mean first `IPCMessages.FileCheckRequest` and after `_setSessionForFile`  if no error with previous.